### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by slovdahl**

### CODE REVIEW

File: tests/git_repo_mock.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def __init__(self, origin, remote_repos):
 
         self.remote_repos = remote_repos
         self._local_repo = GitRepoModel()
-        self._remotes = dict(origin=origin)
+        self._remotes = {"origin": origin}
```
Comment (perlun) [inline tests/git_repo_mock.py:93]: Just out of curiosity, was it so that the `dict` approach is a Python 2 thingie or something?

Comment (slovdahl) [reply tests/git_repo_mock.py:93]: Not sure :thinking: I have seen VS Code complaining about it, but not until this `pylint` update it was actually turned into a failure.

File: tests/test_merge_request.py [hunk 2]
```diff
@@ -216,11 +216,11 @@ def test_accept_remove_branch(self):
...
-                dict(
-                    merge_when_pipeline_succeeds=True,
-                    should_remove_source_branch=boolean,
-                    sha='badc0de',
-                )
+                {
+                    'merge_when_pipeline_succeeds': True,
+                    'should_remove_source_branch': boolean,
+                    'sha': 'badc0de',
+                }
```
Comment (perlun) [inline tests/test_merge_request.py:223]: I'm not 100% convinced that this is really more readable, but if that's what the linters/tooling decree... :grin:

File: marge/job.py [hunk 3]
```diff
@@ -426,14 +426,14 @@ def synchronize_using_gitlab_rebase(self, merge_request, verify_expected_sha=Tru
...
-                )
 
-            return merge_request.sha
+        if verify_expected_sha and merge_request.sha != expected_sha:
+            raise GitLabRebaseResultMismatch(
+                gitlab_sha=merge_request.sha,
+                expected_sha=expected_sha,
+            )
+
+        return merge_request.sha
```
Comment (perlun) [inline marge/job.py:436]: Hmm, what does the old indentation/way of structuring the code even mean in this case? :thinking: `try` with `else`...?

**Review by perlun — APPROVED** [review]

File: marge/job.py (hunk 3 above)
Comment (slovdahl) [reply marge/job.py:436]: https://www.geeksforgeeks.org/try-except-else-and-finally-in-python/

**Pull request merged**